src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -513,4 +525,33 @@ private Result shouldSaveResult(@NotNull Result result, boolean shouldSave) {
             return result;
         }
     }
+
+    /**
+     * Retrieves aggregated feedback details for a given exercise, calculating relative counts based on the total number of distinct results.
+     * The task numbers are assigned based on the associated test case names, using the set of tasks fetched from the database.
+     * <br>
+     * For each feedback detail:
+     * 1. The relative count is calculated as a percentage of the total number of distinct results for the exercise.
+     * 2. The task number is determined by matching the test case name with the tasks.
+     *
+     * @param exerciseId The ID of the exercise for which feedback details should be retrieved.
+     * @return A list of FeedbackDetailDTO objects, each containing:
+     *         - feedback count,
+     *         - relative count (as a percentage of distinct results),
+     *         - detail text,
+     *         - test case name,
+     *         - determined task number (based on the test case name).
+     */
+    public List<FeedbackDetailDTO> findAggregatedFeedbackByExerciseId(long exerciseId) {
+        long distinctResultCount = studentParticipationRepository.countDistinctResultsByExerciseId(exerciseId);
+        Set<ProgrammingExerciseTask> tasks = programmingExerciseTaskService.getTasksWithUnassignedTestCases(exerciseId);
+        List<FeedbackDetailDTO> feedbackDetails = studentParticipationRepository.findAggregatedFeedbackByExerciseId(exerciseId);
+
+        return feedbackDetails.stream().map(detail -> {
+            double relativeCount = (detail.count() * 100.0) / distinctResultCount;
+            int taskNumber = tasks.stream().filter(task -> task.getTestCases().stream().anyMatch(tc -> tc.getTestName().equals(detail.testCaseName()))).findFirst()
+                    .map(task -> tasks.stream().toList().indexOf(task) + 1).orElse(0);
+            return new FeedbackDetailDTO(detail.count(), relativeCount, detail.detailText(), detail.testCaseName(), taskNumber);
+        }).collect(Collectors.toList());
```

Comment:
Architecture test: `no classes should call method Collectors.toList(), because You should use .toList() or .collect(Collectors.toCollection(ArrayList::new)) instead`

-> can you use `.toList()`?